src/components/statusHelper.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import { CheckCircleIcon, TimesCircleIcon, InProgressIcon } from '@patternfly/react-icons';
+import {
+    Flex, FlexItem, FlexModifiers,
+    Button, Tooltip
+} from '@patternfly/react-core';
+
+export const renderStatusIcon = (status) => ({
+    running: <InProgressIcon
+        className="ins-c-remediations-running"
+        aria-label="connection status" />,
+    success: <CheckCircleIcon
+        className="ins-c-remediations-success"
+        aria-label="connection status" />,
+    failure: <TimesCircleIcon
+        className="ins-c-remediations-failure"
+        aria-label="connection status" />
+})[status];
+
+export const renderStatus = (status, text) => ({
+    running: <Flex className='ins-c-remediations-running' breakpointMods={ [{ modifier: FlexModifiers['space-items-sm'] }] }>
+        <FlexItem><b>{ text || 'Running' }</b></FlexItem>
+        <FlexItem><InProgressIcon aria-label="connection status: running"/></FlexItem>
+    </Flex>,
+    success: <Flex className="ins-c-remediations-success" breakpointMods={ [{ modifier: FlexModifiers['space-items-sm'] }] }>
+        <FlexItem><b>{ text || 'Success' }</b></FlexItem>
+        <FlexItem><CheckCircleIcon aria-label="connection status: success"/></FlexItem>
+    </Flex>,
+    failure: <Flex className="ins-c-remediations-failure" breakpointMods={ [{ modifier: FlexModifiers['space-items-sm'] }] }>
+        <FlexItem><b>{ text || 'Failed' }</b></FlexItem>
+        <FlexItem><TimesCircleIcon aria-label="connection status: failed"/></FlexItem>
+    </Flex>
+})[status];
+
+export const statusText = (executorStatus) => ({
+    running: <b className="ins-c-remediations-running"> Running </b>,
+    success: <b className="ins-c-remediations-success"> Suceeded </b>,
+    failure: <b className="ins-c-remediations-failure"> Failed </b>
+})[executorStatus];
+
+export const statusTextPlain = (executorStatus) => ({
+    running: 'Running',
+    success: 'Suceeded',
+    failure: 'Failed'
+})[executorStatus];
+/*
+    executorStatus: bool,
+    systemsStatus: { success: number, failure: number, running: number }
+*/
+export const statusSummary = (executorStatus, systemsStatus, permission, needsTooltip) => {
+    // TODO: Cancel onClick()
+
+    const statusBar = (
+        <Flex className="ins-c-remediations-status-bar">
+            <FlexItem>
+                { statusText(executorStatus) }
+            </FlexItem>
+            <FlexItem>
+                { renderStatus('success', `${systemsStatus.success}`) }
+            </FlexItem>
+            <FlexItem>
+                { renderStatus('failure', `${systemsStatus.failure}`) }
+            </FlexItem>
+            <FlexItem>
+                { renderStatus('running', `${systemsStatus.running}`) }
+            </FlexItem>
+            
+            { permission.permissions.execute && systemsStatus.running > 0 &&
```

Comment:
Instead of `systemsStatus.running` this should be based on executor.status i.e. the button should show up if executor.status is one of [running, pending, acked]

Reply:
Does it mean that we should change `nomalizeStatus` function?
to:
```
export const normalizeStatus = (status) => ({
    running: 'running',
    pending: 'running',
    failure: 'failure',
    canceled: 'failure',
    success: 'success',
    acked: 'running'
})[status];
```